ocfweb/reservations/reserve.py

```diff
@@ -0,0 +1,214 @@
+    )
+
+    verify_contact_email = forms.EmailField(
+        label='Confirm contact e-mail',
+        widget=forms.EmailInput(attrs={'placeholder': 'janedoe@berkeley.edu'}),
+    )
+
+    student_group = forms.CharField(
+        label='Student Group',
+        widget=forms.TextInput(attrs={'placeholder': 'OCF'}),
+        min_length=3,
+        max_length=32,
+    )
+
+    reason = forms.CharField(
+        label='Reason for reservation',
+        widget=forms.TextInput(attrs={'placeholder': ''}),
+    )
+
+    date = forms.DateField(
+        label='Date of reservation (mm/dd/yy)',
+        widget=forms.DateInput(attrs={'placeholder': '02/28/18'}),
+    )
+
+    starttime = forms.TimeField(
+        label='Starting time of reservation (xx:xx)',
+    )
+
+    endtime = forms.TimeField(
+        label='Ending time of reservation (xx:xx)',
+    )
+
+    disclaimer_agreement = forms.BooleanField(
+        label='I agree with the above statement.',
+        error_messages={
+            'required': 'You must agree to our policies.',
+        },
+    )
+
+    def clean_verify_contact_email(self):
+        email = self.cleaned_data.get('contact_email')
+        verify_contact_email = self.cleaned_data.get('verify_contact_email')
+
+        if email and verify_contact_email:
+            if email != verify_contact_email:
+                raise forms.ValidationError("Your emails don't match.")
+        return verify_contact_email
+
+
+@calnet_required
+def request_reservation(request):
+    calnet_uid = request.session['calnet_uid']
+    status = 'new_request'
+
+    # ensure we can even find them in university LDAP
+    # (alumni etc. might not be readable in LDAP but can still auth via CalNet)
+    if not user_attrs_ucb(calnet_uid):
+        return render(
+            request,
+            'reservations/reservations/cant-find-in-ldap.html',
+            {
+                'calnet_uid': calnet_uid,
+                'title': 'Unable to read account information',
+            },
+        )
+
+    if request.method == 'POST':
+        form = RequestForm(request.POST)
+        if form.is_valid():
+            req = NewReservationRequest(
+                real_name=form.cleaned_data['real_name'],
+                contact_email=form.cleaned_data['contact_email'],
+                student_group=form.cleaned_data['student_group'],
+                reason=form.cleaned_data['reason'],
+                date=form.cleaned_data['date'],
+                starttime=form.cleaned_data['starttime'],
+                endtime=form.cleaned_data['endtime'],
+                handle_warnings=NewReservationRequest.WARNINGS_WARN,
+            )
+
+            send_request_to_officers(req)
+            send_request_confirmation(req)
+
+            return render(
+                request,
+                'reservations/pending.html',
+            )
+
+    else:
+        form = RequestForm()
+
+    return render(
```

Comment:
indent this for consistency with the above? or deindent the thing above?

personally i would write this all as one big if-statement

```python
if not user_attrs_ucb(calnet_uid):
    ...
elif request.method == 'POST':
    ...
else: 
    form = RequestForm()
    return render(...)
```

imo this makes the program flow more obvious, as you don't need to check the other branches to see if they actually do return or not

Reply:
I believe the indentation is correct. I can combine the `if not user_attrs_ucb(calnet_uid):` and the `if request.method == 'POST':` into an if/elif/else block if you still desire.

Reply:
it is "correct" in that the flow is correct, but weird because half of your one logical branch is indented differently to the other half